pagseguro.php

```diff
@@ -152,455 +140,437 @@ public function uninstall()
+
+        $requirements['curl'][1] 	= (is_null($requirements['curl'][1])        ? $this->l('Biblioteca cURL instalada.') : $requirements['curl'][1]);
+        $requirements['dom'][1] 	= (is_null($requirements['dom'][1])         ? $this->l('DOM XML instalado.') : $requirements['dom'][1]);
+        $requirements['spl'][1] 	= (is_null($requirements['spl'][1])         ? $this->l('Biblioteca padrão do PHP(SPL) instalada.') : $requirements['spl'][1]);
+        $requirements['version'][1]     = (is_null($requirements['version'][1])     ? $this->l('Versão do PHP superior à 5.3.3.') : $requirements['version'][1]);
+
+        $this->addToView('requirements', $requirements);
+
+        return $this->display(__PS_BASE_URI__ . 'modules/pagseguro', '/views/templates/admin/requirements.tpl');
+
     }
 
-    public function hookPaymentReturn($params)
-    {
-        $this->modulo->returnPaymentConfiguration($params);
-        return $this->display(__PS_BASE_URI__ . 'modules/pagseguro', '/views/templates/hook/payment_return.tpl');
+    private function setContext() {
+        $this->context = Context::getContext();
     }
 
-    private function validatePagSeguroRequirements()
-    {
+    private function validatePagSeguroRequirements() {
         $condional = true;
-        
+
         foreach (PagSeguroConfig::validateRequirements() as $value) {
             if (! Tools::isEmpty($value)) {
                 $condional = false;
                 $this->errors[] = Tools::displayError($value);
             }
         }
-        
+
         if (! $condional) {
             $this->html = $this->displayError(implode('<br />', $this->errors));
         }
-        
+
         return $condional;
     }
 
-    private function createTables()
-    {
-        $sql = 'CREATE TABLE IF NOT EXISTS `' . _DB_PREFIX_ . 'pagseguro_order` (
-            `id` int(11) unsigned NOT NULL auto_increment,
-            `id_transaction` varchar(255) NOT NULL,
-            `id_order` int(10) unsigned NOT NULL ,
-            PRIMARY KEY  (`id`)
-            ) ENGINE=' . _MYSQL_ENGINE_ .
-             ' DEFAULT CHARSET=utf8  auto_increment=1;';
-        
-        if (! Db::getInstance()->Execute($sql)) {
-            return false;
+    private function createTables() {
+
+        $sql = '
+            CREATE TABLE IF NOT EXISTS `' . _DB_PREFIX_ . 'pagseguro_order` (
+                `id` int(11) unsigned NOT NULL auto_increment,
+                `id_transaction` varchar(255) NOT NULL,
+                `id_order` int(10) unsigned NOT NULL,
+                PRIMARY KEY  (`id`)
+            ) ENGINE=' . _MYSQL_ENGINE_ . ' DEFAULT CHARSET=utf8  auto_increment=1;
+        ';
+
+        if (Db::getInstance()->Execute($sql)) {
+            return $this->alterTables();
         }
-        return true;
+
+        return false;
     }
-    
-    private function validatePagSeguroId()
-    {
+
+    private function alterTables() {
```

Comment:
Why do you alter your table directly after their creation ? You should add your columns in the ```CREATE TABLE```